lib/web_ui/lib/src/engine/window.dart

```diff
@@ -49,11 +77,15 @@ class EngineFlutterWindow extends ui.SingletonFlutterWindow {
     engineDispatcher.windows[_windowId] = this;
     engineDispatcher.windowConfigurations[_windowId] = const ui.ViewConfiguration();
     if (_isUrlStrategySet) {
-      _browserHistory =
-          MultiEntriesBrowserHistory(urlStrategy: _customUrlStrategy);
+      _browserHistory = _createDefaultBrowserHistory(_customUrlStrategy);
     }
     registerHotRestartListener(() {
```

Comment:
Does the hot restart teardown the history entries? If it does, we will lose all the url and state even if we memorize the mode. If it does not, I think we can just read the current state to determine whether that is a multi entry or single entry